NGL preliminary documentation

sjbrandenberg opened the pull request:
I put together all of our NGL documentation into a single usecase.md file. It's a lot of content, but I think it works OK as a single file.

sjbrandenberg committed:
Create DesignSafe_Webinar_Oct2021.md

sjbrandenberg committed:
Update and rename usecase.md to usecase_new.md

sjbrandenberg committed:
Update usecase_new.md

sjbrandenberg committed:
Update usecase_new.md

sjbrandenberg committed:
Update usecase_new.md

sjbrandenberg committed:
Update usecase_new.md

sjbrandenberg committed:
Update usecase_new.md

sjbrandenberg committed:
Merge branch 'DesignSafe-CI:master' into master

sjbrandenberg committed:
Update and rename usecase_new.md to usecase.md

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Add files via upload

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Add files via upload

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Delete DesignSafe_Webinar_Oct2021.md

Will merge this with other documentation.

sjbrandenberg committed:
Update usecase.md

Updated heading levels for last three query examples from ## to ###.

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update usecase.md

Updated heading levels for many of the fields

sjbrandenberg committed:
Update usecase.md

Spell check. Liuquefaction is not a word.

kks32 reviewed (approved):
Looks good to me.

You can choose to split into multiple files for installation, set-up, etc...if you like. As long as we point the root to usecase.md so we are consistent.

sjbrandenberg committed:
Update usecase.md

Updating !pip install to correct path to GitHub

sjbrandenberg committed:
Update usecase.md

Updates to reflect new method of connecting to database

kks32 commented on docs/brandenberg/usecase.md:
```diff
@@ -1,89 +1,893 @@
 
 * Please cite [Rathje et al. (2017)](https://doi.org/10.1061/(ASCE)NH.1527-6996.0000246) to acknowledge the use of DesignSafe resources.  
 
 * This software is distributed under the GNU General Public License (https://www.gnu.org/licenses/gpl-3.0.html).  
```
```suggestion
* This software is distributed under the [GNU General Public License](https://www.gnu.org/licenses/gpl-3.0.html).  
```

kks32 commented on docs/brandenberg/usecase.md:
```diff
@@ -1,89 +1,893 @@
+
+<https://nextgenerationliquefaction.org/schema/index.html>
+  
+## Installing Database Connection Script
```
Needs discussion on where this code can be run, i.e., introduce Jupyter

sjbrandenberg replied:
I added a section called "Querying Data via Jupyter Notebooks" that provides this context.

kks32 commented on docs/brandenberg/usecase.md:
```diff
@@ -1,89 +1,893 @@
+you will need to understand the organizational structure of the database, called the schema. The database 
+schema is documented at the following URL:
+
+<https://nextgenerationliquefaction.org/schema/index.html>
```
```suggestion
[https://nextgenerationliquefaction.org/schema/index.html](https://nextgenerationliquefaction.org/schema/index.html)
```

kks32 commented on docs/brandenberg/usecase.md:
```diff
@@ -1,89 +1,893 @@
-> Use case template design
+### Query Wildlife liquefaction array data
 
+This cell queries event information from the EVNT table and surface evidence of liquefaction information from the FLDM table at the Wildlife Array site. The definition of each table and site is below. The query utilizes an INNER JOIN statement to combine tables based on shared keys, and will return all values that have matching keys in both tables. For more details, see <https://www.w3schools.com ...
```
```suggestion
This cell queries event information from the `EVNT` table and surface evidence of liquefaction information from the `FLDM` table at the Wildlife Array site. The definition of each table and site is below. The query utilizes an `INNER JOIN` statement to combine tables based on shared keys, and will return all values that have matching keys in both tables. For more details, see [https://www.w3schools.com/sql/sql_join_inner.asp](https://www.w3schools.com/sql/sql_join_inner.asp)
```

kks32 commented on docs/brandenberg/usecase.md:
````diff
@@ -1,89 +1,893 @@
+import pandas as pd
+```
+
+command = 'SELECT (SELECT COUNT(SCPG_ID) FROM SCPG) as "CPT Soundings", '
+command += '(SELECT COUNT(BORH_ID) FROM BORH) as "Boreholes", '
+command += '(SELECT COUNT(GSWG_ID) FROM GSWG) as "Surface Wave Measurements", '
+command += '(SELECT COUNT(GINV_ID) FROM GINV) as "Invasive VS Measurements", '
+command += '(SELECT COUNT(FLDM_ID) FROM FLDM WHERE FLDM_SFEV=1) as "Liquefaction Observations", '
+command += '(SELECT COUNT(FLDM_ID) FROM FLDM WHERE FLDM_SFEV=0) as "Non-Liquefaction Observations"'
+
+scpg_count = ngl.read_sql(command).T
+scpg_count.columns=["Count"]
+scpg_count
````
Add a code section using ```

sjbrandenberg replied:
I have removed all code blocks from the documentation

kks32 reviewed (changes requested):
Please add a ## Background section at the start and other suggested changes

sjbrandenberg committed:
Update docs/brandenberg/usecase.md

Co-authored-by: Krishna Kumar <3963513+kks32@users.noreply.github.com>

sjbrandenberg committed:
Update docs/brandenberg/usecase.md

Co-authored-by: Krishna Kumar <3963513+kks32@users.noreply.github.com>

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update usecase.md

Moved code documentation from here to the notebooks, and significantly reduced the length of this documentation file.

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update usecase.md

sjbrandenberg committed:
Update docs/brandenberg/usecase.md

Co-authored-by: Krishna Kumar <3963513+kks32@users.noreply.github.com>

kks32 committed:
Merge branch 'master' into master
